src/components/study-container.jsx

```diff
@@ -313,10 +316,18 @@ export function StudyContainer({ view, onChangeTab }) {
             ws.onmessage = function (event) {
                 const eventData = JSON.parse(event.data);
                 const updateTypeHeader = eventData.headers[UPDATE_TYPE_HEADER];
+                const computationTypeHeader = eventData.headers[COMPUTATION_TYPE_HEADER];
                 if (updateTypeHeader === 'STUDY_ALERT') {
                     sendAlert(eventData);
                     return; // here, we do not want to update the redux state
                 }
+                // used to update the computation type
+                if (
+                    updateTypeHeader === NotificationType.COMPUTATION_PARAMETERS &&
+                    isValidComputingType(computationTypeHeader)
+                ) {
+                    dispatch(setStudyParamsChanged(computationTypeHeader));
```

Comment:
Instead of adding a new entry in redux store, you should use the existing one I think :
`dispatch(studyUpdated(eventData));`
One Websocket, one state in the store, multiple consumers which filter interesting notifications by HEADERS values.

Reply:
I thought about it at the beginning but I was not sure since the name studyUpdated refers to study modification. But yes it works fine.